StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/FileHeaderCodeFixProvider.cs

```diff
@@ -106,8 +140,32 @@ private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, Styl
                 }
             }
 
+            // Remove copyright lines in reverse order
+            removalList.Reverse();
+            foreach (var triviaLine in removalList)
+            {
+                trivia = trivia.RemoveAt(triviaLine);
+            }
+
             string newLineText = document.Project.Solution.Workspace.Options.GetOption(FormattingOptions.NewLine, LanguageNames.CSharp);
-            return root.WithLeadingTrivia(CreateNewHeader(document.Name, settings, newLineText).Add(SyntaxFactory.CarriageReturnLineFeed).Add(SyntaxFactory.CarriageReturnLineFeed).AddRange(trivia));
+
+            var newHeaderTrivia = CreateNewHeader(document.Name, settings, newLineText);
+            if (!isMalformedHeader && copyrightTriviaIndex.HasValue)
+            {
+                // Replace copyright element in place.
+                return root.WithLeadingTrivia(trivia.ReplaceRange(trivia[copyrightTriviaIndex.Value], newHeaderTrivia));
+            }
+            else
+            {
+                // Add blank line if we don't already have comments at top of file.
+                if ((trivia.Count == 0) || (trivia[0].Kind() != SyntaxKind.SingleLineCommentTrivia))
+                {
+                    newHeaderTrivia = newHeaderTrivia.Add(SyntaxFactory.CarriageReturnLineFeed);
+                }
+
+                // Insert header at top of the file.
+                return root.WithLeadingTrivia(newHeaderTrivia.Add(SyntaxFactory.CarriageReturnLineFeed).AddRange(trivia));
+            }
```

Comment:
Ah see what you mean now. You're talking about 

```C#
   // <author>Peter</author>
   // <copyright>FooBar</copyright>
   // <summary>Stuff<summary>
```

with the old code would have been turned into

```C#
// <copyright>FooBar</copyright>
```

but now looks like this

```C#
   // <author>Peter</author>
// <copyright>FooBar</copyright>
   // <summary>Stuff<summary>
```